src/GprTool/Program.cs

```diff
@@ -334,26 +334,41 @@ protected override Task OnExecute(CommandLineApplication app)
                     return Task.CompletedTask;
                 }
 
-                var ownerAndRepositoryName = Repository
-                    .Replace("\\", "/")
-                    .Split("/", StringSplitOptions.RemoveEmptyEntries)
-                    .Take(2)
-                    .ToList();
+                var (githubOwner, githubRepositoryName, githubRepositoryUri) = Repository.BuildGithubRepositoryDetails();
 
-                if (ownerAndRepositoryName.Count != 2)
+                if (githubOwner == null
+                    || githubRepositoryName == null
+                    || githubRepositoryUri == null 
+                    || githubRepositoryUri.Host != "github.com")
                 {
                     Console.WriteLine(
                         "Invalid repository value. Please use the following format: owner/repository. E.g: jcansdale/gpr");
                     return Task.CompletedTask;
                 }
 
-                owner = ownerAndRepositoryName[0];
-                var repositoryName = ownerAndRepositoryName[1];
-                var repositoryUrl = $"https://github.com/{owner}/{repositoryName}";
+                owner = githubOwner;
 
-                if (NuGetUtilities.ShouldRewriteNupkg(PackageFile, repositoryUrl, nuGetVersion))
+                if (NuGetUtilities.ShouldRewriteNupkg(PackageFile, githubRepositoryUri.ToString(), nuGetVersion))
                 {
-                    rewrittenPackageFile = NuGetUtilities.RewriteNupkg(PackageFile, repositoryUrl, nuGetVersion);
+                    rewrittenPackageFile = NuGetUtilities.RewriteNupkg(PackageFile, githubRepositoryUri.ToString(), nuGetVersion);
+                }
+            }
+            else
+            {
+                var manifest = NuGetUtilities.ReadNupkgManifest(PackageFile);
+                var manifestRepositoryUrl = manifest.Metadata.Repository?.Url;
+                
+                var (githubOwner, githubRepositoryName, githubRepositoryUri) = manifestRepositoryUrl.BuildGithubRepositoryDetails();
+
+                if (githubOwner == null
+                    || githubRepositoryName == null
+                    || githubRepositoryUri == null 
+                    || githubRepositoryUri.Host != "github.com")
+                {
+                    Console.WriteLine($"Package is missing a valid <RepositoryUrl /> XML element value: {manifestRepositoryUrl} " +
```

Comment:
Technically it's the project (.csproj) that is missing a `RepositoryUrl` rather than the .nupkg/.nuspec (which uses `<repository url="..." />`).

```suggestion
                    Console.WriteLine($"Project is missing a valid <RepositoryUrl /> XML element value: {manifestRepositoryUrl} " +
```